stacks/portstack.go

```diff
@@ -376,29 +376,41 @@ func (ps *PortStack) HandleEth(dst []byte) (n int, err error) {
 //
 // If a handler returns any other error the port is closed.
 func (ps *PortStack) handleEth(dst []byte) (n int, err error) {
+	//this would be much better named as getNextOutboundPacket
+
+	//println ("portStack:handleEth ",string(dst)) //@@@REMOVE
+
 	switch {
 	case len(dst) < int(ps.mtu):
+		
 		return 0, io.ErrShortBuffer
 
 	case !ps.IsPendingHandling():
+		
 		return 0, nil // No remaining packets to handle.
 	}
 	n = ps.arpClient.handle(dst)
 	if n != 0 {
 		return n, nil
 	}
 
+	//i don't know why this is defined within the scope of handleEth
 	type Socket interface {
 		Close()
 		IsPendingHandling() bool
-		HandleEth(dst []byte) (int, error)
+		 HandleEth(dst []byte) (int, error)
 	}
-
+	//or why this is an anonymous function is stored in a local variable before invocation  - I can see it narrows scope, but it's an odd pattern and confusing as the code does not run in the order it appears
```

Comment:
We can extract both Socket and handleSocket to package level in this or a future PR. 

Note: Socket would have to be unexported, so `socket`,`isocket` or something.

Reply:
done .. Socket is now socket, and handleSocket is now package level - will test before next PR